plugins/secrets-manager-aws-secret-manager/src/static/startup/providers/secrets/secretsManagerFactory.ts

```diff
@@ -0,0 +1,45 @@
+import { ConfigService } from "@nestjs/config";
+import {
+  GetSecretValueCommand,
+  SecretsManagerClient
+} from "@aws-sdk/client-secrets-manager"
+import { Logger } from "@nestjs/common"
+import { Secrets } from "./secrets";
+
+export const secretsManagerFactory = {
+  provide: "AWS_SECRETS_MANAGER",
+  useFactory: async (configService: ConfigService): Promise<Record<string, string>> => {
+    const logger = new Logger()
+
+    const client = new SecretsManagerClient({
+      region: configService.get("AWS_REGION")
+    })
+
+    var secrets: Record<string, string> = {}
+
+    for (const [name, path] of Object.entries(Secrets)) {
+      const [secret_id, secret_name] = path.split(":")
+
+      try {
+        const response = await client.send(new GetSecretValueCommand({ SecretId: secret_id }))
+
+        const secrets_list: Record<string, string> = JSON.parse(
+          response.SecretString ? response.SecretString :
+            response.SecretBinary ? new TextDecoder().decode(response.SecretBinary) :
+            "{}"
+        )
+
+        secrets = {
+          ...secrets,
+          ...(secret_name ? { [name]: secrets_list[secret_name] } : secrets_list)
+        }
+      } catch(err) {
+        logger.warn(`Error while loading secret named "${secret_name}" from secret id "${secret_id}" - ${err.message}`, "SecretsManager")
```

Comment:
Be careful as we use ts. We will need to check for the error type otherwise we will get a build error like
<img width="1056" alt="image" src="https://github.com/amplication/plugins/assets/2861984/a89168f9-e7e3-494f-902d-17ce82813872">

I'd also suggest to log as ERROR instead of WARN as these might need immediate attention from the user